ArduCopter/commands.cpp

```diff
@@ -106,8 +106,8 @@ bool Copter::set_home(const Location& loc, bool lock)
 bool Copter::far_from_EKF_origin(const Location& loc)
 {
     // check distance to EKF origin
-    const struct Location &ekf_origin = inertial_nav.get_origin();
-    if (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M) {
+    Location ekf_origin;
+    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M)) {
```

Comment:
This is technically a behavior change, but I'm going to argue that it's innocuous. What previously happened is that if the origin wasn't available we zero'd the location, which meant we compared the passed in location to 0, 0. But since we are trying to check that the distance from the location to (0, 0) is less then 50 km, this will almost always fail anyways. And for the non GPS case, we actually do require people to set the origin via MAVLink which would allow this to pass. So while this technically changes the behavior I considered it acceptable. If the change isn't okay, then I can bring back the zero logic.

This also applies to sub.